web3/_utils/request.py

```diff
@@ -23,15 +24,34 @@ def _remove_session(key: str, session: requests.Session) -> None:
 def _get_session(*args: Any, **kwargs: Any) -> requests.Session:
     cache_key = generate_cache_key((args, kwargs))
     if cache_key not in _session_cache:
-        _session_cache[cache_key] = requests.Session()
+        session = requests.Session()
+        adapter = requests.adapters.HTTPAdapter(**extract_adapter_kwargs(**kwargs))
+
+        session.mount('http://', adapter)
+        session.mount('https://', adapter)
+
+        _session_cache[cache_key] = session
     return _session_cache[cache_key]
 
 
 def make_post_request(endpoint_uri: URI, data: bytes, *args: Any, **kwargs: Any) -> bytes:
     kwargs.setdefault('timeout', 10)
-    session = _get_session(endpoint_uri)
+    session = _get_session(endpoint_uri, **kwargs)
     # https://github.com/python/mypy/issues/2582
-    response = session.post(endpoint_uri, data=data, *args, **kwargs)  # type: ignore
+    response = session.post(endpoint_uri, data=data, *args,
+                            **extract_request_kwargs(**kwargs))  # type: ignore
     response.raise_for_status()
 
     return response.content
+
+
+def extract_adapter_kwargs(**kwargs: Any) -> Dict[str, Any]:
+    adapter_args = ['pool_connections', 'pool_maxsize', 'max_retries', 'pool_block']
```

Comment:
I'd agree that this isn't the most maintainable design, i'd try playing around with these helper functions to automatically detect the available parameters - similar to [how it's done here](https://stackoverflow.com/questions/55590419/filter-keyword-arguments-when-calling-function).